### PR TITLE
JIRA# ROS-150: Remove entry from Ob and put it in TblObject where it belongs.  Also …

### PR DESCRIPTION
**Pull request by ghost**
…implement Location.fetch

### CODE REVIEW

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 1]
```diff
@@ -93,4 +91,42 @@ object Location {
 
       case LTLimbo => StoreFail
     }
+
+  def fetch(loc: Location, k: Ctxt, globalEnv: TblObject): Ob =
+    loc.genericType match {
+      case LTCtxtRegister(reg) =>
+        if (reg < k.reg.size) {
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: In the C++ implementation `k.reg.size` is the constant `NumberOfCtxtRegs` which is 10.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 2]
```diff
@@ -93,4 +91,42 @@ object Location {
...
+
+      case LTGlobalVariable(offset) =>
+        if (offset < globalEnv.numberOfSlots()) {
+          globalEnv.slot(offset)
+        } else {
+          Ob.INVALID
+        }
+
+      case LTBitField(ind, level, offset, spanSize) =>
+        k.env.getField(ind, level, offset, spanSize)
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: We don't need the `sign` field?

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 3]
```diff
@@ -93,4 +91,42 @@ object Location {
...
+
+      case LTBitField00(offset, spanSize) =>
+        k.env.getField(0, 0, offset, spanSize)
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: We don't need the `sign` field?

**Review by tymm — COMMENTED** [review]
LGTM but there are three lines of code where I would like to get more insight before approving.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 1 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: I'd prefer to leave it that way; I suspect the compiler will replace it anyway if the implementation uses a constant for the size.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 2 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: Hrm.  There's no sign field when we store it, so I need to dig into what's going on here.

File: rosette-scala/src/main/scala/coop/rchain/rosette/TblObject.scala [hunk 4]
```diff
@@ -1,6 +1,6 @@
 package coop.rchain.rosette
 
-case class TblObject(override val entry: Seq[Ob],
+case class TblObject(val entry: Seq[Ob],
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/TblObject.scala]: No `val` needed here.

**Review by tymm — COMMENTED** [review]
Added comment.

**Review by tymm — APPROVED** [review]

**Pull request merged**